mobile/native/src/trade/order.rs

```diff
@@ -0,0 +1,44 @@
+use crate::api_model;
+use crate::common::ContractSymbol;
+use crate::common::Direction;
+use uuid::Uuid;
+
+// When naming this the same as `api_model::order::OrderType` the generated code somehow uses
+// `trade::OrderType` and contains errors, hence different name is used.
+// This is likely a bug in frb.
+#[derive(Debug, Clone, Copy)]
+pub enum OrderTypeTrade {
+    Market,
+    Limit { price: f64 },
+}
+
+#[derive(Debug, Clone, Copy)]
+pub enum OrderStatus {
+    Open,
+    Filled,
```

Comment:
Can you add a comment explaining these states please?

Reply:
I added all the states we defined yesterday from #149 and rustdoc for them. Note that I introduced separate states for `Rejected` (orderbook rejects submission) and `Failed` (trade execution failed) in the state machine, so we can properly distinguish them. Note that not all internal states are exposed to the UI via the API.

I added the rustdoc on the business logic class, because that one is more exhaustive in states and represents the actual state machine.